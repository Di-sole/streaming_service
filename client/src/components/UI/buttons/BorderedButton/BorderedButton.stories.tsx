import type { Meta, StoryObj } from '@storybook/react';
import BorderedButton from './BorderedButton';
import './BorderedButton.module.scss';

const meta: Meta<typeof BorderedButton> = {
    title: 'Button/BorderedButton',
    component: BorderedButton,
    tags: ['autodocs'],
    argTypes: {
        size: {
            type: 'string',
            descriptrion: 'Размер кнопки',
            defaultValue: 'medium',
            options: ['large', 'medium', 'small'],
        },
    },
    decorators: [
        (Story) => (
            <div
                style={{
                    background: '#100e19',
                    height: '50px',
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'center',
                }}
            >
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof BorderedButton>;

export const LargeButton: Story = {
    decorators: [
        (Story) => (
            <div style={{ width: '100px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        size: 'large',
        children: 'Button',
    },
};

export const MediumButton: Story = {
    args: {
        size: 'medium',
        children: 'Button',
    },
};

export const SmallButton: Story = {
    args: {
        size: 'small',
        children: 'Button',
    },
};
