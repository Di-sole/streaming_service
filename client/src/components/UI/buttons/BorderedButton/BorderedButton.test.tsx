import { render, screen } from '@testing-library/react';
import BorderedButton from './BorderedButton';

describe('BORDEREDBUTTON SNAPSHOTS TESTS', () => {
    test('size="large" BorderedButton should not be changed', () => {
        render(<BorderedButton className="button" size="large" children="Text" />);
        const button = screen.getByRole('button');

        expect(button).toMatchSnapshot();
    });

    test('size="medium" BorderedButton should not be changed', () => {
        render(<BorderedButton className="button" size="medium" children="Text" />);
        const button = screen.getByRole('button');

        expect(button).toMatchSnapshot();
    });

    test('size="small" BorderedButton should not be changed', () => {
        render(<BorderedButton className="button" size="small" children="Text" />);
        const button = screen.getByRole('button');

        expect(button).toMatchSnapshot();
    });
});
