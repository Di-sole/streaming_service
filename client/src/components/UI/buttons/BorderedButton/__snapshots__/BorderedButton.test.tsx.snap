// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BORDEREDBUTTON SNAPSHOTS TESTS size="large" BorderedButton should not be changed 1`] = `
<button
  class="button button_large button"
  type="button"
>
  Text
</button>
`;

exports[`BORDEREDBUTTON SNAPSHOTS TESTS size="medium" BorderedButton should not be changed 1`] = `
<button
  class="button button_medium button"
  type="button"
>
  Text
</button>
`;

exports[`BORDEREDBUTTON SNAPSHOTS TESTS size="small" BorderedButton should not be changed 1`] = `
<button
  class="button button_small button"
  type="button"
>
  Text
</button>
`;
